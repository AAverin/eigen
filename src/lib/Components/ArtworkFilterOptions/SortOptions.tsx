import React from "react"
import { NavigatorIOS } from "react-native"
import { SingleSelectOptionScreen } from "./SingleSelectOption"

interface SortOptionsScreenProps {
  navigator: NavigatorIOS
}

export const SortOptionsScreen: React.SFC<SortOptionsScreenProps> = ({ navigator }) => {
  return <SingleSelectOptionScreen filterType="sort" filterText="Sort" navigator={navigator} />
}
