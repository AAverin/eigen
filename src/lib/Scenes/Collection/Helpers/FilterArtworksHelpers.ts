import { Aggregations, FilterArray } from "lib/utils/ArtworkFiltersStore"
import { forOwn, omit } from "lodash"

// General filter types and objects
export enum FilterParamName {
  sort = "sort",
  medium = "medium",
  priceRange = "priceRange",
  size = "dimensionRange",
  color = "color",
  gallery = "partnerID",
  institution = "partnerID",
  timePeriod = "majorPeriods",
  waysToBuyBuy = "acquireable",
  waysToBuyBid = "atAuction",
  waysToBuyInquire = "inquireableOnly",
  waysToBuyMakeOffer = "offerable",
}

// Types for the parameters passed to Relay
interface FilterParams {
  sort?: string
  medium?: string
  priceRange?: string
  dimensionRange?: string
  color?: string
  partnerID?: string
  majorPeriods?: string
  acquireable?: boolean
  inquireableOnly?: boolean
  atAuction?: boolean
  offerable?: boolean
}

export enum FilterDisplayName {
  sort = "Sort",
  medium = "Medium",
  priceRange = "Price range",
  size = "Size",
  color = "Color",
  gallery = "Gallery",
  institution = "Institution",
  timePeriod = "Time period",
  waysToBuy = "Ways to buy",
}

export interface InitialState {
  initialState: {
    selectedFilters: FilterArray
    appliedFilters: FilterArray
    previouslyAppliedFilters: FilterArray
    applyFilters: boolean
    aggregations: Aggregations
  }
}

export interface AggregateOption {
  displayText: string
  paramValue: string
}

const defaultFilterParams = {
  sort: "-decayed_merch",
  medium: "*",
  priceRange: "*-*",
  dimensionRange: "*-*",
  atAuction: false,
  acquireable: false,
  inquireableOnly: false,
  offerable: false,
} as FilterParams

const paramsFromAppliedFilters = (appliedFilters: FilterArray, filterParams: FilterParams) => {
  appliedFilters.forEach(appliedFilterOption => {
    // @ts-ignore STRICTNESS_MIGRATION
    filterParams[appliedFilterOption.paramName] = appliedFilterOption.paramValue
  })

  return filterParams
}

export const filterArtworksParams = (appliedFilters: FilterArray) => {
  return paramsFromAppliedFilters(appliedFilters, { ...defaultFilterParams })
}

const getChangedParams = (appliedFilters: FilterArray) => {
  const filterParams = paramsFromAppliedFilters(appliedFilters, {})

  // when filters cleared return default params
  return Object.keys(filterParams).length === 0 ? defaultFilterParams : filterParams
}

export const changedFiltersParams = (
  currentFilterParams: any /* STRICTNESS_MIGRATION */,
  selectedFilterOptions: FilterArray
) => {
  const selectedFilterParams = getChangedParams(selectedFilterOptions)
  const changedFilters = {}

  /** If a filter option has been updated e.g. was { medium: "photography" } but
   *  is now { medium: "sculpture" } add the updated filter to changedFilters. Otherwise,
   *  add filter option to changedFilters.
   */
  forOwn(getChangedParams(selectedFilterOptions), (_value, filterType) => {
    // @ts-ignore STRICTNESS_MIGRATION
    if (currentFilterParams[filterType] === selectedFilterParams[filterType]) {
      const omitted = omit(selectedFilterParams, [filterType])
      // @ts-ignore STRICTNESS_MIGRATION
      if (omitted[filterType]) {
        // @ts-ignore STRICTNESS_MIGRATION
        changedFilters[filterType] = omitted[filterType]
      }
    } else {
      // @ts-ignore STRICTNESS_MIGRATION
      changedFilters[filterType] = selectedFilterParams[filterType]
    }
  })

  return changedFilters
}
