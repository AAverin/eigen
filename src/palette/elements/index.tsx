export * from "./Cards"
export * from "./Touchable"
export * from "./Typography"
