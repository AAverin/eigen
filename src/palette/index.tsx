export * from "./atoms"
export * from "./organisms"
export * from "./elements"
export * from "./Theme"
