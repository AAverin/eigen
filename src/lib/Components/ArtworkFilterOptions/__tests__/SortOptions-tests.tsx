import { Theme } from "@artsy/palette"
import { FakeNavigator as MockNavigator } from "lib/Components/Bidding/__tests__/Helpers/FakeNavigator"
import React from "react"
import * as renderer from "react-test-renderer"
import { SortOptionsScreen as SortOptions } from "../SortOptions"

describe("Sort Options Screen", () => {
  let mockNavigator: MockNavigator

  beforeEach(() => {
    mockNavigator = new MockNavigator()
  })

  it("renders properly", () => {
    const SortOptionsScreen = renderer.create(
      <Theme>
        <SortOptions navigator={mockNavigator as any} />
      </Theme>
    )
    expect(SortOptionsScreen).toMatchSnapshot()
  })
})
