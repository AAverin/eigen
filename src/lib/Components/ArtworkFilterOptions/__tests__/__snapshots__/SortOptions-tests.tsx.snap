// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Options Screen renders properly 1`] = `
<View
  flexGrow={1}
  style={
    Array [
      Object {
        "display": "flex",
        "flexGrow": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 20,
        },
      ]
    }
  >
    <View
      alignItems="flex-end"
      mb={2}
      mt={0.5}
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "marginBottom": 20,
            "marginTop": 5,
          },
        ]
      }
    >
      <View
        ml={2}
        mt={2}
        onTouchStart={[Function]}
        style={
          Array [
            Object {
              "marginLeft": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="18px"
          bbWidth="18px"
          className="Iconios__Icon-z4d8jr-0 cjvLsm"
          fill="black100"
          focusable={false}
          height="18px"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          vbHeight={18}
          vbWidth={18}
          width="18px"
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M12.06 15.94L5.12 9l6.94-6.94.88.88L6.88 9l6.06 6.06z"
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={0}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      fontFamily="Unica77LL-Medium"
      fontSize="16px"
      lineHeight="26px"
      mt={2}
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 16,
            "lineHeight": 26,
            "marginTop": 20,
          },
          Object {},
        ]
      }
    >
      Sort
    </Text>
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          "Default",
          "Price (low to high)",
          "Price (high to low)",
          "Recently Updated",
          "Recently Added",
          "Artwork year (descending)",
          "Artwork year (ascending)",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Default
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Price (low to high)
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Price (high to low)
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Recently Updated
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Recently Added
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Artwork year (descending)
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#E5E5E5",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0.5,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                flexDirection="row"
                flexGrow={1}
                justifyContent="space-between"
                p={2}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "space-between",
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                    },
                  ]
                }
              >
                <Text
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="16px"
                  lineHeight="24px"
                  style={
                    Array [
                      Object {
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  Artwork year (ascending)
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E5E5E5",
          "display": "flex",
          "flexGrow": 1,
        },
      ]
    }
  />
</View>
`;
